### ISSUE TITLE
miri can't run single tests

### ISSUE THREAD

**Issue by djugei** [rust-lang/miri#1414]
I can't seem to find a way to make miri run just a subset of my tests.
cargo miri test always runs all of them and i don't see a way to cut that down.

**Comment by vakaras** [comment]
Have you tried: `cargo miri test -- -- <test-name>` ?

**Comment by RalfJung** [comment]
@djugei it would be helpful to know what you tried.

Also see the docs at https://github.com/rust-lang/miri#using-miri.

**Comment by djugei** [comment]
i tried cargo miri test -- test-name, but not the double-double-dashes, which work.

literally had to go /second on the docs to find it, there isn't an example of the double-double-dash-syntax. i also read the help text (no info) and the src/bin/miri.rs to have a look at comand line parsing, but thats hand-rolled, uses global state, and i did not look into the deeper function calls.

**Comment by RalfJung** [comment]
Clearly our docs need improvement, you are not the only one to be confused by this. FWIW, the right file to look at would have been `src/bin/cargo-miri.rs`, but even there this is not easy to see (Cc https://github.com/rust-lang/miri/issues/546).

As a start, I'll add the example back to the README... we used to have one I think, but it somehow disappeared.

(Also, it's not *global* state, we are not that *that* crazy.^^)

**Comment by RalfJung** [comment]
e help text *does* indicate both double-dashes:
```
Usage:
    cargo miri [subcommand] [options] [--] [<miri opts>...] [--] [<program opts>...]
```
Looks like this is still not clear enough, maybe we should have examples here, too.

**Comment by djugei** [comment]
right, i looked at cargo-miri.rs, misremembered.

and you are right, the help text is there, must have overlooked it, guess i expected it to be right after the test-subcommand, also cargo miri --help and cargo miri test --help produce the same output (not more specific for subcommand)

**Comment by djugei** [comment]
i think with those changes i would have found what i was looking for

**Comment by RalfJung** [comment]
> i think the main confusion for me is that its different from how cargo without miri works

Yeah, that's a bit weird. But I don't really see another way as we also need a way to pass flags to Miri.

Maybe we could re-expose all these flags through `cargo miri` directly so they do not need a `--`. Or maybe that should be an env var, `MIRIFLAGS` or so. Though at this point we'd also have a backwards compatibility hazard if we changed this. :/
I opened an issue just for this: https://github.com/rust-lang/miri/issues/1416

Also not sure if you saw that, I opened a PR to add more examples to the docs: https://github.com/rust-lang/miri/pull/1415
